build/commands/lib/util.js

```diff
@@ -689,6 +689,9 @@ const util = {
       await util.buildNativeRedirectCC()
 
       const extraGnGenOpts = config.extraGnGenOpts ? [config.extraGnGenOpts] : []
+      if (config.isCI) {
+        extraGnGenOpts.push('--check')
```

Comment:
confirmed this flag is persisted in `build.ninja` for regenerating purposes:
```
rule gn
  command = ../../buildtools/win/gn.exe --root=../.. -q --check --regeneration gen .
  pool = console
  description = Regenerating ninja files
```

this means the auto-regeneration will be performed with the `--check` enabled, which is what we want.